docs/grammar.txt

```diff
@@ -127,7 +127,7 @@ StringLiteral = '"' ([^"\r\n\\] | '\\' .)* '"'
 Identifier = [a-zA-Z_$] [a-zA-Z_$0-9]*
 
 HexNumber = '0x' [0-9a-fA-F]+
-DecimalNumber = [0-9]+
+DecimalNumber = [0-9]+ ( '.' [0-9]+ )? ( [eE] [0-9]+ )?
```

Comment:
This is used in describing the fixed point type so perhaps one of them should be using a different rule.

Alternatively we could remove the elevated role of built in types and just consider everything a user type in the grammar.

Reply:
Good catch! For now, I will change `Fixed ` and `Ufixed` to use a plain `[0-9]+`. Then we can discuss removing built-in types. The problem with that is that parsers will be unable to distinguish UserDefinedTypeNames and will also accept built-in types as identifiers.

Reply:
Just wondering if we need `DecimalNumber` and `HexNumber` as a different rule since it is only used within `NumberLiteral`? Though we can argue about that in a separate PR.

> The problem with that is that parsers will be unable to distinguish UserDefinedTypeNames and will also accept built-in types as identifiers.

That's true, though not sure if that must be handled within the parser or the semantics. So far we tried to handle a lot in the grammar.

Reply:
Another remark: `1.` and `1.e10` are valid according to the compiler, the grammar should reflect that. We can start a separate discussion to argue whether that is good or bad if needed.

Reply:
Hmm... hadn't noticed that. Fixed the grammar.